src/transformation/visitors/variable-declaration.ts

```diff
@@ -13,16 +13,19 @@ import { transformPropertyName } from "./literal";
 
 export function transformArrayBindingElement(
     context: TransformationContext,
-    name: ts.ArrayBindingElement | ts.Expression
+    name: ts.ArrayBindingElement
 ): lua.Identifier {
     if (ts.isOmittedExpression(name)) {
         return lua.createAnonymousIdentifier(name);
     } else if (ts.isIdentifier(name)) {
         return transformIdentifier(context, name);
-    } else if (ts.isBindingElement(name) && ts.isIdentifier(name.name)) {
+    } else if (ts.isBindingElement(name)) {
+        // TODO: It should always be true when called from `transformVariableDeclaration`,
```

Comment:
The assertion below is invalid in the case of `for (const [[x]] of string.gmatch('', '')) {}`

Reply:
Well, handle it somehow - `assert` is meant to never fail under the expected conditions. I haven't looked much into it, but in case of `transformVariableDeclaration` it's only used in the optimization transform, so maybe the same thing should be done in the other case